cpp/unreal_plugins/Vehicle/Source/Vehicle/VehiclePawn.cpp

```diff
@@ -0,0 +1,138 @@
+    skeletal_mesh_component_->SetCanEverAffectNavigation(false);
+
+    RootComponent = skeletal_mesh_component_;
+
+    // Setup vehicle movement
+    vehicle_movement_component_ = CreateDefaultSubobject<UVehicleMovementComponent>(TEXT("AVehiclePawn::vehicle_movement_component_"));
+    SP_ASSERT(vehicle_movement_component_);
+    vehicle_movement_component_->SetIsReplicated(true); // Enable replication by default
+    vehicle_movement_component_->UpdatedComponent = skeletal_mesh_component_;
+    // this ensures that the body doesn't ever sleep. Need this to bypass a Chaos bug that doesn't take torque inputs to wheels into consideration for determining the sleep state of the body.
+    vehicle_movement_component_->SleepThreshold = 0;
+
+    // Setup camera
+    FVector camera_location(
+        Config::get<float>("VEHICLE.VEHICLE_PAWN.CAMERA_COMPONENT.POSITION_X"),
+        Config::get<float>("VEHICLE.VEHICLE_PAWN.CAMERA_COMPONENT.POSITION_Y"),
+        Config::get<float>("VEHICLE.VEHICLE_PAWN.CAMERA_COMPONENT.POSITION_Z"));
+
+    FRotator camera_orientation(
+        Config::get<float>("VEHICLE.VEHICLE_PAWN.CAMERA_COMPONENT.PITCH"),
+        Config::get<float>("VEHICLE.VEHICLE_PAWN.CAMERA_COMPONENT.YAW"),
+        Config::get<float>("VEHICLE.VEHICLE_PAWN.CAMERA_COMPONENT.ROLL"));
+
+    camera_component_ = CreateDefaultSubobject<UCameraComponent>(TEXT("AVehiclePawn::camera_component_"));
+    SP_ASSERT(camera_component_);
+
+    camera_component_->SetRelativeLocationAndRotation(camera_location, camera_orientation);
+    camera_component_->SetupAttachment(skeletal_mesh_component_);
+    camera_component_->bUsePawnControlRotation = false;
+    camera_component_->FieldOfView = Config::get<float>("VEHICLE.VEHICLE_PAWN.CAMERA_COMPONENT.FOV");
+
+    // Setup IMU sensor
+    FVector imu_location(
+        Config::get<float>("VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.POSITION_X"),
+        Config::get<float>("VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.POSITION_Y"),
+        Config::get<float>("VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.POSITION_Z"));
+
+    FRotator imu_orientation(
+        Config::get<float>("VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.PITCH"),
+        Config::get<float>("VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.YAW"),
+        Config::get<float>("VEHICLE.VEHICLE_PAWN.IMU_COMPONENT.ROLL"));
+
+    imu_component_ = CreateDefaultSubobject<UBoxComponent>(TEXT("AVehiclePawn::imu_component_"));
+    ASSERT(imu_component_);
+
+    imu_component_->SetRelativeLocationAndRotation(imu_location, imu_orientation);
+    imu_component_->SetupAttachment(skeletal_mesh_component_);
+}
+
+AVehiclePawn::~AVehiclePawn()
+{
+    SP_LOG_CURRENT_FUNCTION();
+}
+
+void AVehiclePawn::SetupPlayerInputComponent(UInputComponent* input_component)
+{
+    SP_ASSERT(input_component);
+    APawn::SetupPlayerInputComponent(input_component);
+}
+
+// Apply the drive torque in[N.m] to the vehicle wheels.The applied driveTorque persists until the
```

Comment:
You don't need any of the functions below. You can inline these functions directly in `VehicleAgent` since they're only called in one place. Fix the flawed spacing in the comments.